File: 1267.统计参与通信的服务器.c
/*
 * @lc app=leetcode.cn id=1267 lang=c
 *
 * [1267] 统计参与通信的服务器
 */

 // @lc code=start


int countServers(int** grid, int gridSize, int* gridColSize) {
    int n = gridSize, m = gridColSize[0];
    int rows[n], cols[m];
    memset(rows, 0, sizeof(rows));
    memset(cols, 0, sizeof(cols));
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < m; j++) {
            if (grid[i][j] == 1) {
                rows[i]++;
                cols[j]++;
            }
        }
    }

    int ans = 0;
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < m; j++) {
            if (grid[i][j] == 1 && (rows[i] >= 2 || cols[j] >= 2)) {
                ans++;
            }
        }
    }
    return ans;
}
// @lc code=end

File: 1267.统计参与通信的服务器.go
package leetcode

/*
 * @lc app=leetcode.cn id=1267 lang=golang
 *
 * [1267] 统计参与通信的服务器
 */

// @lc code=start
func countServers(grid [][]int) int {
	row := make([]int, len(grid))
	column := make([]int, len(grid[0]))
	for i := 0; i < len(grid); i++ {
		for j := 0; j < len(grid[0]); j++ {
			if grid[i][j] == 1 {
				row[i]++
				column[j]++
			}
		}
	}

	result := 0
	for i := 0; i < len(grid); i++ {
		for j := 0; j < len(grid[0]); j++ {
			if grid[i][j] == 1 && (row[i] >= 2 || column[j] >= 2) {
				result++
			}
		}
	}

	return result
}

// @lc code=end
